Shell/NewCNF.cpp

```diff
@@ -1244,43 +1245,46 @@ void NewCNF::process(QuantifiedFormula* g, Occurrences &occurrences)
 
 void NewCNF::processBoolterm(TermList ts, Occurrences &occurrences)
 {
-  CALL("NewCNF::process(TermList)");
+  CALL("NewCNF::processBoolterm");
 
   if (ts.isVar()) {
     processBoolVar(POSITIVE, ts.var(), occurrences);
     return;
   }
 
   Term* term = ts.term();
-  ASS_REP(term->isSpecial(), term->toString());
+  //ASS_REP(term->isSpecial(), term->toString());
```

Comment:
So, processBoolterm may now be called on non-bool  terms (which don't need any processing)?

Reply:
Not quite, `processBoolterm` can be called on a term of type `$o` that is not a special.

Reply:
Only called on Bool terms, but in some cases, the term may not be a special. If this is case, we are guaranteed that it will be a constant or variable. 

As an alternative, we could mark a problem as higher-order as soon as we read the `thf` tag in parsing which would mean that NewCNF would never be called on a quasi-THF problem